languages/csharp/exercises/concept/chars/.docs/instructions.md

```diff
@@ -0,0 +1,56 @@
+In this exercise you will implement a partial set of utility routines to help a developer
+clean up identifier names.
+
+In the first 4 tasks you will gradually build up the routine `Clean` A valid identifier comprises
+zero or more letters and underscores.
+
+The final task is a little more esoteric involving inserting a character into an identifier
+
+In all cases the input string is guaranteed to be non-null.
+
+### 1. Replace any spaces encountered with underscores
+
+This also applies to leading and trailing spaces
+
+```csharp
+Identifier.Clean("my   Id");
+// => "my___Id"
+```
+
+### 2. Replace control characters with the upper case string "CTRL"
+
+```csharp
+Identifier.Clean("my\0Id");
+// => "myCTRLId",
+```
+
+### 3. Convert kebab-case to camel-case
+
+An identifier such as my-object becomes myObject
+
+```csharp
+Identifier.Clean("à-ḃç");
+// => "àḂç"
+```
+
+### 4. Ensure that an already valid identifier remains unchnged by the routine
+
+Note that the `Clean` method should treat an empty string as valid
+
+```csharp
+Identifier.Clean("àḃç");
+// => "àḃç"
+```
+
+### 5. Insert character into a string
```

Comment:
I've been thinking about this for a bit. The thing you want to have the student know about is that characters are ordered, right? As in, one character is smaller or greater than another. I was thinking if perhaps we could also make this part of the `Clean` method. Something along the line of removing an "unfriendly" character that was added to an identifier by accident. Another option I thought of was to exclude a certain "range" of numbers, which would involve comparing characters against a minimum and/or maximum character to check if it is in a range.

Reply:
What about excluding greek letters `ch >= 'α' && ch <= 'ω'`?  This is not great but slightly more justifiable than excluding some digits but allowing ohers.  I don't see any way to retrieve the block or script to which which a character belongs and if students are using UTF-16 numeric codes then I'm sure they wouldn't be phased by a simple comparison.

Reply:
I think excluding greek letters is an excellent idea!